components/brave_new_tab_ui/components/default/braveToday/cards/PublisherMeta.tsx

```diff
@@ -4,21 +4,200 @@
+  font-size: 12px;
+
+  &:hover,
+  &:focus {
+    background-color: ${p => p.theme.color.contextMenuHoverBackground};
+    color: ${p => p.theme.color.contextMenuHoverForeground};
+  }
+
+  &:active {
+    // TODO(petemill): Theme doesn't have a context menu interactive color,
+    // make one and don't make entire element opaque.
+    opacity: .8;
+  }
+
+  &:focus-visible {
+    outline: solid 1px ${p => p.theme.color.brandBrave};
+  }
+`
+
+export default function PublisherMetaComponent (props: Props) {
+
+  const [isMenuOpen, setIsMenuOpen] = React.useState(false)
+
+  const triggerElementRef = React.useRef<HTMLElement>(null)
+
+  const onClickCloseMenu = React.useCallback((e: MouseEvent) => {
+    const triggerElement = triggerElementRef.current
+    if (!triggerElement || triggerElement.contains(e.target as Node)) {
+      return
+    }
+    setIsMenuOpen(false)
+  }, [setIsMenuOpen])
+
+  const onKeyDown = React.useCallback((e: KeyboardEvent) => {
+    if (e.defaultPrevented) {
+      return
+    }
+    if (e.key === 'Escape') {
+      setIsMenuOpen(false)
+    }
+  }, [setIsMenuOpen])
+
+  const toggleMenu = React.useCallback((e: React.MouseEvent) => {
+    e.stopPropagation()
+    e.preventDefault()
+    setIsMenuOpen((value) => !value)
+  }, [setIsMenuOpen])
+
+  // Setup or remote event listeners when opens or closes
+  // or this element is removed.
+  React.useEffect(() => {
+    const removeEventListeners = () => {
+      window.removeEventListener('click', onClickCloseMenu)
+      window.removeEventListener('keydown', onKeyDown)
+    }
+    if (!isMenuOpen) {
+      removeEventListeners()
+    } else {
+      // TODO(petemill): set element focus when using keyboard arrows.
+      window.addEventListener('click', onClickCloseMenu)
+      window.addEventListener('keydown', onKeyDown)
+    }
+    return removeEventListeners
+  }, [isMenuOpen])
+
+  const onClickDisablePublisher = React.useCallback(() => {
+    props.onSetPublisherPref(
+      props.publisher.publisher_id,
+      false
+    )
+  }, [props.publisher, props.onSetPublisherPref])
+
+  const commandText = React.useMemo<string>(() => {
+    const raw = getLocale('braveTodayDisableSourceCommand')
+    const publisherIndex = raw.indexOf('$1')
+    if (publisherIndex === -1) {
+      console.warn('Locale string for braveTodayDisableSourceCommand did not have a $1 replacement area for publisher name.', raw)
+      return `${raw} ${props.publisher.publisher_name}`
+    }
+    return raw.substr(0, publisherIndex) +
+      props.publisher.publisher_name +
+      raw.substr(publisherIndex + 2)
+  }, [props.publisher.publisher_name])
+
   return (
-    <Component>
-      {props.publisher.publisher_name}
-    </Component>
+    <PublisherMeta>
+      <Trigger
+        className={isMenuOpen ? isOpenClassName : undefined}
+        onClick={toggleMenu}
+        innerRef={triggerElementRef}
+        aria-haspopup='true'
+        aria-expanded={isMenuOpen ? 'true' : 'false'}
```

Comment:
nit: I might be wrong but won’t this translate appropriately if you just set the property to the variable? Ex it handles bool to string conversion